client/ayon_core/hosts/hiero/api/plugin.py

```diff
@@ -604,31 +609,42 @@ def load(self):
         return track_item
 
 
-class Creator(LegacyCreator):
+class Creator(NewCreator):
     """Creator class wrapper
     """
     clip_color = "Purple"
     rename_index = None
 
-    def __init__(self, *args, **kwargs):
-        super(Creator, self).__init__(*args, **kwargs)
-        import ayon_core.hosts.hiero.api as phiero
-        self.presets = get_current_project_settings()[
-            "hiero"]["create"].get(self.__class__.__name__, {})
 
+    def apply_settings(self, project_settings):
+        resolve_create_settings = (
+            project_settings.get("hiero", {}).get("create")
+        )
+        self.presets = resolve_create_settings.get(
+            self.__class__.__name__, {}
+        )
+
```

Comment:
I would say that default implementation works a little bit better, it does override attributes from settings on the object.
```suggestion
```

Reply:
Yes, but it does not store it in `presets` but instead writes the content into actual attributes on the class, so if settings defines e.g. `hiero.create.CLASSNAME.hello = 1` then your class will end up with `class.hello = 1`.

So usually class attributes are defined at the top with their defaults, but the default `apply_settings` overrides them from settings if they are available there, e.g. like `clip_color = "Purple" is currently defined at the top.